jax/experimental/sparse/ad.py

```diff
@@ -66,9 +66,16 @@ def reconstruct(i, grad_out):
       f_recons = jax.vmap(f_recons)
     return f_recons(grad_out)
 
+  def tree_is_none(tree):
+    return tree_util.tree_all(tree_util.tree_map(lambda x: x is None, tree))
```

Comment:
Since `None` represents an empty subtree and not a leaf, a more efficient way to do this is `tree_is_none = lambda tree: not jax.tree.leaves(tree)`